source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2515 @@
+      - session: { id: session0, client: client0 }
+      - session: { id: session1, client: session0 }
+
+
+Test Format
+-----------
+
+Each specification test file can define one or more tests, which inherit some
+top-level configuration (e.g. namespace, initial data). YAML and JSON test files
+are parsed as a document by the test runner. This section defines the top-level
+keys for that document and links to various sub-sections for definitions of
+nested structures (e.g. individual `test`_, `operation`_).
+
+Although test runners are free to process YAML or JSON files, YAML is the
+canonical format for writing tests. YAML files may be converted to JSON using a
+tool such as `js-yaml <https://github.com/nodeca/js-yaml>`__ .
+
+
+Top-level Fields
+~~~~~~~~~~~~~~~~
+
+The top-level fields of a test file are as follows:
+
+- ``description``: Required string. The name of the test file.
+
+  This SHOULD describe the common purpose of tests in this file and MAY refer to
+  the filename (e.g. "updateOne-hint").
+
+.. _schemaVersion:
+
+- ``schemaVersion``: Required string. Version of this specification to which the
+  test file complies. Test runners will use this to determine compatibility
+  (i.e. whether and how the test file will be interpreted). The format of this
+  string is defined in `Version String`_; however, test files SHOULD NOT need to
+  refer to specific patch versions since patch-level changes SHOULD NOT alter
+  the structure of the test format (as previously noted in `Schema Version`_).
+
+.. _runOn:
+
+- ``runOn``: Optional array of documents. List of server version and/or topology
+  requirements for which the tests in this file can be run. If no requirements
+  are met, the test runner MUST skip this test file.
+
+  If set, the array should contain at least one document. The structure of each
+  document is defined in `runOnRequirement`_.
+
+.. _createEntities:
+
+- ``createEntities``: Optional array of documents. List of entities (e.g.
+  client, collection, session objects) that should be created before each test
+  case is executed. The structure of each document is defined in `entity`_.
+
+  Test files SHOULD define entities in dependency order, such that all
+  referenced entities (e.g. client) are defined before any of their dependent
+  entities (e.g. database, session).
+
+.. _initialData:
+
+- ``initialData``: Optional array of documents. Data that should exist in
+  collections before each test case is executed.
+
+  If set, the array should contain at least one document. The structure of each
+  document is defined in `collectionData`_. Before each test and for each
+  `collectionData`_, the test runner MUST drop the collection and insert the
+  specified documents (if any) using a "majority" write concern. If no documents
+  are specified, the test runner MUST create the collection with a "majority"
+  write concern.
+
+.. _tests:
+
+- ``tests``: Required array of documents. List of test cases to be executed
```

Comment:
I'm generally sensitive to the term `document` referred to as a specific object type, versus a response / result from a MongoDB collection, here I'd suggest "Required array of Test objects" where `Test` is a link to a Test object.

Reply:
I originally used "document" because the test files are typically parsed as Extended JSON, but I agree that it'd be preferable to refer to "object" unless we're actually referring to a MongoDB document. Made this change throughout the file.